examples/httpbin-demo/src/charm.py

```diff
@@ -32,43 +31,44 @@ class HttpbinDemoCharm(ops.CharmBase):
 
     def __init__(self, framework: ops.Framework):
         super().__init__(framework)
+        framework.observe(self.on.collect_unit_status, self._on_collect_status)
         framework.observe(self.on['httpbin'].pebble_ready, self._on_httpbin_pebble_ready)
         framework.observe(self.on.config_changed, self._on_config_changed)
+        self.log_level = str(self.config['log-level'])  # str() is here for the type checker.
+        self.container = self.unit.get_container('httpbin')
+
+    def _on_collect_status(self, event: ops.CollectStatusEvent):
+        """Report the status of the workload (runs after each event)."""
+        if self.log_level.lower() not in VALID_LOG_LEVELS:
+            event.add_status(ops.BlockedStatus(f"invalid log level: '{self.log_level}'"))
+            return
```

Comment:
If we go with my more radical suggestion, this wouldn't be needed. Or if we think it would be better we could remove the `errors=`, catch the `ValueError` in `__init__` and then also have here something like:

```python
try:
    self.load_config(HttpBinConfig)
except ValueError as e:
    event.add_status(ops.BlockedStatus(str(e)))
```